module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/MakeLinkTest.php

```diff
@@ -0,0 +1,233 @@
+<?php
+/**
+ * makeLink view helper Test Class
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2019.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+namespace VuFindTest\View\Helper\Root;
+
+use VuFind\Record\Router;
+use VuFind\View\Helper\Root\MakeLink;
+
+/**
+ * makeLink view helper Test Class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class MakeLinkTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Get MakeLink helper with mock view
+     *
+     * return \Laminas\View\Helper\EscapeHtml
+     */
+    protected function getHelper()
+    {
+        $escapeHtml = new \Laminas\View\Helper\EscapeHtml();
+        $view = $this->createMock(\Laminas\View\Renderer\PhpRenderer::class);
+        $view->method('plugin')->will($this->returnValue($escapeHtml));
```

Comment:
I would add, in front of `->will`, something like `->expects($this->once())->with($this->equalTo('escapeHtmlAttr'))`, just to be sure we're returning this when we expect to.